app/javascript/tests/subject/MoveToHousehold.test.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { shallow } from 'enzyme';
+import { Button, Modal, Form } from 'react-bootstrap';
+import ReactTooltip from 'react-tooltip';
+import MoveToHousehold from '../../components/subject/MoveToHousehold.js'
+import { mockPatient1 } from '../mocks/mockPatients'
+
+const authyToken = "Q1z4yZXLdN+tZod6dBSIlMbZ3yWAUFdY44U06QWffEP76nx1WGMHIz8rYxEUZsl9sspS3ePF2ZNmSue8wFpJGg==";
+
+function getWrapper(patient) {
+    return shallow(<MoveToHousehold patient={patient} authenticity_token={authyToken} />);
+}
+
+function getPatientName(patient) {
+  return `${patient.first_name} ${patient.middle_name} ${patient.last_name}`;
+}
+
+describe('MoveToHousehold', () => {
+    it('Properly renders Move To Household button', () => {
+        const wrapper = getWrapper(mockPatient1);
+        expect(wrapper.find(Button).length).toEqual(1);
+        expect(wrapper.find(Button).text().includes('Move To Household')).toBeTruthy();
+        expect(wrapper.find('i').hasClass('fa-house-user')).toBeTruthy();
+        expect(wrapper.find(Button).prop('disabled')).toBeFalsy();
+        expect(wrapper.find(ReactTooltip).exists()).toBeFalsy();
+    });
+
+    it('Clicking the Move to Household button opens modal and modal renders correctly', () => {
+      const wrapper = getWrapper(mockPatient1);
+      const toggleModalSpy = jest.spyOn(wrapper.instance(), "toggleModal");
+      const updateTableSpy = jest.spyOn(wrapper.instance(), "updateTable");
+      // Necessary for the spy to be called on click
+      wrapper.instance().forceUpdate();
+
+      expect(toggleModalSpy).toHaveBeenCalledTimes(0);
+      expect(updateTableSpy).toHaveBeenCalledTimes(0);
+      expect(wrapper.state('showModal')).toBe(false);
+      expect(wrapper.find(Modal).exists()).toBe(false);
+
+      wrapper.find(Button).simulate('click');
+
+      expect(toggleModalSpy).toHaveBeenCalledTimes(1);
+      expect(updateTableSpy).toHaveBeenCalledTimes(1);
+      expect(wrapper.state('showModal')).toBe(true);
+      expect(wrapper.find(Modal).exists()).toBe(true);
+      expect(wrapper.find(Modal).find(Form.Label).text()).toEqual(
+        `Please select the new monitoree that will respond for ${getPatientName(mockPatient1)}.`
+      )
+      expect(wrapper.find(Modal).find(Form.Label).find('b').text()).toEqual(getPatientName(mockPatient1));
+      expect(wrapper.find(Modal).find('p').text())
+      .toEqual(
+        'You may select from the provided existing Head of Households and monitorees who are self reporting.' + 
+        `${getPatientName(mockPatient1)} will be immediately moved into the selected monitoree's household.`
+      );
```

Comment:
Add checks that the search bar, custom table and cancel button are there

Reply:
Added, though running into an issue with it finding two instances of an element that should only render once when doing a `mount`git  instead of a `shallow`. Added a TODO for now but will try to resolve.
1fba2ea

Reply:
the reason you see that is `mount` fully renders all children components, so something like `<Form.Control type="input" id="some_id" />` when you mount render will have the child `<input>` with the same id as well as the form control.  That's why I try to use `shallow` mostly unless there is some weird children/parent stuff I need to test

Reply:
updated here: https://github.com/SaraAlert/SaraAlert/pull/531/commits/b1f04abd405f230fca597fc31ec6d1652fefb39a